samples/sid_end_device/boards/nrf52840dk_nrf52840.overlay

```diff
@@ -4,35 +4,7 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
-&spi3 {
-	status = "disabled";
-};
-
-&spi2 {
-	status = "disabled";
-};
-
```

Comment:
if we bring this back, and revert deletion of bus_nrf_spi.c we could still have the nrfx implementation on nrf52840 dk.

We could add some kconfig warning that if someone build with the semetech shield on this platform this is not supported.

but I think that by simply reverting deletion of this file and deletion of nrfx_spi.c we could retain the old working behaviour on the nrf52 platform